### PR TITLE
rm test train inference config_optimizier + modif test.yml

### PR DESCRIPTION
**Pull request by clemoule**
rm test.py, train.py, inference.py (replaced by fit test predict)
rm configure optimizer
modify test.yml

### CODE REVIEW

**Review by cbovalo — DISMISSED** [review]

File: .github/workflows/tests.yml
Comment (LBerth) [inline .github/workflows/tests.yml:1]: is it possible to put some comments in the file to explain what each line tests ? Like 'basic fit', 'resume training', etc ... But I don't know if the file format will allow it

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -47,15 +47,12 @@ jobs:
...
-        coverage run -p bin/main.py fit --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.fast_dev_run False --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --ckpt_path /home/runner/work/py4cast/py4cast/lightning_logs/version_test/checkpoints/last.ckpt
-        coverage run -p bin/main.py test --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.fast_dev_run False --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --ckpt_path /home/runner/work/py4cast/py4cast/lightning_logs/version_test/checkpoints/last.ckpt
-        
+        coverage run -p bin/main.py fit --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --trainer.logger.init_args.version version_test
```
Comment (LBerth) [inline .github/workflows/tests.yml:51]: Can you also first do a fit in fast_dev_run mode ? in case some stuff doesn't work when the logger and checkpoint is not available

File: bin/inference.py
Comment (LBerth) [inline bin/inference.py:1]: Have you checked that when using `main.py predict` the grib file is correctly saved like here ? and can we still choose the date of the predict ? and change the nb of steps ?
@CorentinSeznec do you have other requirements that we should check, some stuff that this script did ?

File: bin/train.py [hunk 2]
```diff
@@ -1,390 +0,0 @@
...
-if nb_nodes > 1:
-    gpus_per_node = len(os.environ.get("SLURM_STEP_GPUS", "1").split(","))
-    global_rank = int(os.environ.get("SLURM_PROCID", 0))
-    local_rank = global_rank - gpus_per_node * (global_rank // gpus_per_node)
-    print(
-        f"Global rank: {global_rank}, Local rank: {local_rank}, Gpus per node: {gpus_per_node}"
-    )
-    os.environ["LOCAL_RANK"] = str(local_rank)
-    os.environ["GLOBAL_RANK"] = os.environ.get("SLURM_PROCID", 0)
-    os.environ["NODE_RANK"] = os.environ.get("SLURM_NODEID", 0)
```
Comment (LBerth) [inline bin/train.py:44]: Could you try one multinode training and see if all GPUs are working ? same on one node and 4 GPUs. If it doesn't work we'll have to check the lightning docs to see how it can manage multi-node

File: bin/train.py [hunk 3]
```diff
@@ -1,390 +0,0 @@
...
-        )
-    }
-    loggers.update(mlflow_logger)
-
-print(
-    "--> Model, checkpoints, and tensorboard artifacts "
-    + f"will be saved in {save_path}."
-)
-
-loggers["TensorBoardLogger"].experiment.add_custom_scalars(layout)
```
Comment (LBerth) [inline bin/train.py:286]: I'd like to put this back somewhere, maybe in the lightning module. It allows to have a graph in tensorboard, with overlapping train and valid loss. To see if we are overfitting

File: bin/inference.py
Comment (CorentinSeznec) [reply bin/inference.py:1]: Yes, nb_step, the date and the choice of the floating point precision is important. I think the grib saving needs a lightning_module.precision = 32.
Maybe the dataset_conf ? I don't know how the cli take this into account.

And the inference script uses this file: config/IO/poesy_grib_settings.json to know where and how to save the gribs. You can put it in a yaml file if it is easier for you.

I think that is all it needs.

File: config/CLI/trainer.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # lightning.pytorch==2.4.0
 seed_everything: true
 
-ckpt_path: #EMPTY: runs as usual. NOT EMPTY: load weights and hyperparameters from checkpoint
+ckpt_path: lightning_logs/version_1/checkpoints/epoch=03-val_mean_loss=129.19.ckpt #EMPTY: runs as usual. NOT EMPTY: load weights and hyperparameters from checkpoint
```
Comment (cbovalo) [inline config/CLI/trainer.yaml:4]: Maybe you want to keep ckpt_path empty for the PR...

File: py4cast/datasets/base.py [hunk 5]
```diff
@@ -407,10 +407,10 @@ def is_valid(self) -> bool:
...
                 file_format=self.settings.file_format,
             ):
-                print(
-                    f"invalid: {self.timestamps.validity_times[0]}, \
-                        {self.accessor.parameter_namer(param)}. Check file."
-                )
+                # print(
+                #     f"invalid: {self.timestamps.validity_times[0]}, \
+                #         {self.accessor.parameter_namer(param)}. Check file."
+                # )
```
Comment (cbovalo) [inline py4cast/datasets/base.py:413]: Why keeping these commented lines ?

File: py4cast/lightning.py [hunk 6]
```diff
@@ -628,15 +628,40 @@ def on_load_checkpoint(self, checkpoint):
...
-        return self.forward(batch)
+        preds = self.forward(batch)
+        if not hasattr(self, 'all_preds'):
+            self.all_preds = []
+        self.all_preds.append(preds)
+        return preds
+
+    def on_predict_end(self):
+        """
+        Méthode appelée à la fin de la phase de prédiction.
```
Comment (cbovalo) [inline py4cast/lightning.py:647]: In english ;)

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -47,15 +47,12 @@ jobs:
...
-        coverage run -p bin/main.py fit --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.fast_dev_run False --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --ckpt_path /home/runner/work/py4cast/py4cast/lightning_logs/version_test/checkpoints/last.ckpt
-        coverage run -p bin/main.py test --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.fast_dev_run False --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --ckpt_path /home/runner/work/py4cast/py4cast/lightning_logs/version_test/checkpoints/last.ckpt
-        
+        coverage run -p bin/main.py fit --config config/CLI/trainer.yaml --config config/CLI/dataset/dummy.yaml --config config/CLI/model/unetrpp.yaml --trainer.max_epochs 1 --data.batch_size 1 --trainer.limit_train_batches 1 --trainer.logger.version version_test
```
Comment (LBerth) [inline .github/workflows/tests.yml:51]: add a fit with `trainer.fast_dev_run=True` please

File: py4cast/lightning.py [hunk 8]
```diff
@@ -628,15 +638,42 @@ def on_load_checkpoint(self, checkpoint):
...
+            self.all_preds.append(preds)
+        return preds
+
+    def on_predict_end(self):
+        if self.dataset_name == "poesy":
+            self.grib_writing()
+
+    def grib_writing(self):
+        with open(self.io_conf, "r") as f:
+            save_settings = GribSavingSettings.schema().loads((f.read()))
```
Comment (LBerth) [inline py4cast/lightning.py:663]: you can use json load and pass the dict as keywork arguments

**Review by cbovalo — APPROVED** [review]
